api/src/main/java/org/pmiops/workbench/access/AccessModuleServiceImpl.java

```diff
@@ -0,0 +1,138 @@
+          .put(AccessModule.RAS_LINK_LOGIN_GOV, AccessModuleName.RAS_LOGIN_GOV)
+          .put(AccessModule.DATA_USE_AGREEMENT, AccessModuleName.DATA_USER_CODE_OF_CONDUCT)
+          .put(AccessModule.PUBLICATION_CONFIRMATION, AccessModuleName.PUBLICATION_CONFIRMATION)
+          .put(AccessModule.PROFILE_CONFIRMATION, AccessModuleName.PROFILE_CONFIRMATION)
+          .build();
+
+  private static final BiMap<BypassTimeTargetProperty, AccessModuleName>
+      AUDIT_TO_STORAGE_ACCESS_MODULE =
+          ImmutableBiMap.<BypassTimeTargetProperty, AccessModuleName>builder()
+              .put(BypassTimeTargetProperty.ERA_COMMONS_BYPASS_TIME, AccessModuleName.ERA_COMMONS)
+              .put(
+                  BypassTimeTargetProperty.COMPLIANCE_TRAINING_BYPASS_TIME,
+                  AccessModuleName.RT_COMPLIANCE_TRAINING)
+              .put(
+                  BypassTimeTargetProperty.TWO_FACTOR_AUTH_BYPASS_TIME,
+                  AccessModuleName.TWO_FACTOR_AUTH)
+              .put(BypassTimeTargetProperty.RAS_LINK_LOGIN_GOV, AccessModuleName.RAS_LOGIN_GOV)
+              .put(
+                  BypassTimeTargetProperty.DATA_USE_AGREEMENT_BYPASS_TIME,
+                  AccessModuleName.DATA_USER_CODE_OF_CONDUCT)
+              .build();
+
+  @Autowired
+  public AccessModuleServiceImpl(
+      Provider<List<DbAccessModule>> dbAccessModulesProvider,
+      Clock clock,
+      UserAccessModuleDao userAccessModuleDao,
+      UserDao userDao,
+      UserServiceAuditor userServiceAuditor,
+      Provider<WorkbenchConfig> configProvider) {
+    this.dbAccessModulesProvider = dbAccessModulesProvider;
+    this.clock = clock;
+    this.userAccessModuleDao = userAccessModuleDao;
+    this.userDao = userDao;
+    this.userServiceAuditor = userServiceAuditor;
+    this.configProvider = configProvider;
+  }
+
+  @Override
+  public void updateBypassTime(long userId, AccessModule accessModuleName, boolean isBypassed) {
+    DbAccessModule accessModule =
+        getDbAccessModuleFromApi(dbAccessModulesProvider.get(), accessModuleName);
+    if (!accessModule.getBypassable()) {
+      throw new ForbiddenException("Bypass: " + accessModuleName.toString() + " is not allowed.");
+    }
+    final DbUser user = userDao.findUserByUserId(userId);
+    final List<DbUserAccessModule> dbUserAccessModules = userAccessModuleDao.getAllByUser(user);
+    Optional<DbUserAccessModule> retrievedAccessModule =
+        dbUserAccessModules.stream()
+            .filter(m -> m.getAccessModule().getName().equals(accessModule.getName()))
+            .findFirst();
+    final Timestamp newBypassTime =
+        isBypassed ? new Timestamp(clock.instant().toEpochMilli()) : null;
+    final Timestamp previousBypassTime =
+        retrievedAccessModule.map(DbUserAccessModule::getBypassTime).orElse(null);
+    final DbUserAccessModule userAccessModuleToUpdate;
+
+    logger.info(
+        String.format(
+            "Bypassing %s(uid: %d) for module %s",
```

Comment:
please update this log to use the username, and whether isBypassed is true.  Maybe use "Bypassing" or "Unbypassing" here